memory/src/main/scala/filodb.memory/format/vectors/Histogram.scala

```diff
@@ -316,17 +320,18 @@ final case class MaxHistogram(innerHist: MutableHistogram, max: Double) extends
       // find rank for the quantile using total number of occurrences (which is the last bucket value)
       var rank = q * topBucketValue
       // using rank, find the le bucket which would have the identified rank
-      val b = firstBucketGTE(rank)
+      val bucketNum = firstBucketGTE(rank)
 
       // now calculate quantile.  No need to special case top bucket since we will always cap top at max
-      if (b == 0 && bucketTop(0) <= 0) return bucketTop(0)
+      if (bucketNum == 0 && bucketTop(0) <= 0) return bucketTop(0)
       else {
         // interpolate quantile within le bucket
-        var (bucketStart, bucketEnd, count) = (0d, Math.min(bucketTop(b), max), bucketValue(b))
-        if (b > 0) {
-          bucketStart = bucketTop(b-1)
-          count -= bucketValue(b-1)
-          rank -= bucketValue(b-1)
+        var (bucketStart, bucketEnd, count) = (Math.max(0d, min),
```

Comment:
Not a chance I completely understand what's going on here, but off the top of my head: what if `bucketNum = 0`, `le = 0.001`, and `min = 99999999`? Wouldn't the interpolated value be huge in that case?

Reply:
This would be a case of bad instrumentation from user then. (since they can't expect to configure bucket that low and emit min with such a huge value) The idea is that we need to improve the accuracy of quantile calculation by extrapolating between [min, max] range

Reply:
Some context : the min and max is also published by the user as part of histogram ( this are not query time inputs )

Reply:
> since they can't expect to configure bucket that low and emit min with such a huge value

What about a bucketing schema like this?:
```
le = 1, 2, 4, 8 , 16
```
And the histogram accounts for a single value: `2`. Doesn't that cause the same scenario? I.e.:
```
bucketNum = 0 -> le = 1
min = 2 -> min > le
```

Reply:
I see -- it looks like you're right that this state:
> bucketNum = 0, le = 0.001, and min = 99999999

would never happen.